src/main/java/org/springframework/guice/module/SpringModule.java

```diff
@@ -109,21 +121,84 @@ private void bind(ConfigurableListableBeanFactory beanFactory) {
 				Provider<?> namedProvider = BeanFactoryProvider.named(beanFactory,
 						beanName, type);
 				if (!type.isInterface() && !ClassUtils.isCglibProxyClass(type)) {
-					bindConditionally(binder(), name, type, typeProvider, namedProvider);
+					bindConditionally(binder(), name, type, typeProvider, namedProvider, bindingAnnotation);
 				}
 				for (Class<?> iface : ClassUtils.getAllInterfacesForClass(type)) {
-					bindConditionally(binder(), name, iface, typeProvider, namedProvider);
+					bindConditionally(binder(), name, iface, typeProvider, namedProvider, bindingAnnotation);
 				}
 				for (Type iface : type.getGenericInterfaces()) {
-					bindConditionally(binder(), name, iface, typeProvider, namedProvider);
+					bindConditionally(binder(), name, iface, typeProvider, namedProvider, bindingAnnotation);
 				}
 			}
 		}
 	}
 
+	private Optional<Annotation> getAnnotationForBeanDefinition(BeanDefinition definition) {
+		if (definition instanceof AnnotatedBeanDefinition
+				&& ((AnnotatedBeanDefinition) definition).getFactoryMethodMetadata() != null) {
+			try {
+				Method factoryMethod = getFactoryMethod(beanFactory, definition);
+				return Arrays.stream(AnnotationUtils.getAnnotations(factoryMethod))
+						.filter(a -> Annotations.isBindingAnnotation(a.annotationType())).findFirst();
+			} catch (Exception e) {
+				return Optional.empty();
+			}
+		} else {
+			return Optional.empty();
+		}
+	}
+
+	private Method getFactoryMethod(ConfigurableListableBeanFactory beanFactory,
+			BeanDefinition definition) throws Exception {
```

Comment:
@dsyer I had to copy this from Boot's `BeanTypeRegistry`. Any suggestions on a better way of handling this?

Reply:
If it works, I say ship it. Or are you still missing some of the metadata?

Reply:
Works, just thought I might be missing some helpful utility class. Thanks!